client/src/document/ingredients/prose.tsx

```diff
@@ -1,4 +1,4 @@
-import { DisplayH2, DisplayH3 } from "./utils";
+import { DisplayH2, DisplayH3, DisplayH4 } from "./utils";
```

Comment:
Please see if you can do something like this instead:

```js
function DisplayHeader({level, id, title, titleAsText}) {
  const Tag = `h${level}`;
  return <Tag id={id}>{title}</Tag>
}
```
basically a rewrite of all the `DisplayH2`, `DisplayH3`, `DisplayH4` functions into one. 
And when you call this you do something like this:

```jsx
<DisplayHeading level={section.isH4 ? 4 : section.isH3 ? 3 : 2} id={id} title={section.title} titleAsText={section.titleAsText}/>
```
```

Reply:
Great suggestion! This is working nicely.

Reply:
An alternative might be to use `as` (e.g. `<div as="h3" />`)?

Reply:
That is a good suggestion too @Ryuno-Ki. When using the following with `Tag = 'h{level}'` its throwing an error as we need the `Permalink` component nested in there too.

```js

Type '{ children: Element; id: string; }' is not assignable to type 'IntrinsicAttributes'.
  Property 'children' does not exist on type 'IntrinsicAttributes'.  TS2322

    66 |   const Tag = `h${level}`;
    67 |   return (
  > 68 |     <Tag id={id.toLowerCase()}>
       |      ^
    69 |       <Permalink title={title} titleAsText={titleAsText} id={id} />
    70 |     </Tag>
    71 |   )
```

Using a separate functional component for `Tag` and returning `<div as={`h${level}`}` doesn't play too nicely either. Still figuring out what path will be most ideal.

Reply:
Looks like since we are in TypeScript land defining a dynamic JSX element like `const Tag = 'h' + level` will do some weird stuff with JSX.IntrinsicAttributes probably due to the way its created. I found a workaround by using 
``` 
const Tag = `h${level}` as keyof JSX.IntrinsicElements; 
```
this way we can nest the `<Permalink />` component inside `<Tag>..</Tag>` and then our `DisplayHeading` component works like a charm. One alternative to this typescript workound would be to directly call `React.createElement` instead of directly going to JSX from a string.

Reply:
I'm happy to announce, after condensing each of the DisplayH2,H3,H4 functions into a single DisplayHeading component, everything is working as it did before the changes. Sections are split by h2,h3 OR h4 as like expected.